### PR TITLE
[Feat] Add line bot webhook.

### PR DESCRIPTION
**Pull request by moojing**
Add a web hook for the line bot, and pass testing data for now.

### Related Issue
- https://github.com/Yukaii/zuzugo/issues/87

### CODE REVIEW

File: lib/notification/lineBot.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+// @todo: pass rent data from api.
+const message: TextMessage = {
+  type: 'text',
+  text: 'Zuzugo got a new house!',
+};
+
+// @todo: get groupId from database.
+export default function sendLineBotMessage(groupId: string) {
+  client.pushMessage(groupId, message)
```
Comment (Yukaii) [inline lib/notification/lineBot.ts]: 要 await 或 return promise

File: pages/api/line/events.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    const {
+      userId = '',
+      groupId = '',
+      roomId = '',
+      // @todo: save chatType to database.
+      // type: chatType,
+    } = firstEvent.source as User & Group & Room;
+
+    const chatId = userId || groupId || roomId;
+    sendLineBotMessage(chatId)
```
Comment (Yukaii) [inline pages/api/line/events.ts]: 要 await

**Review by Yukaii — CHANGES_REQUESTED** [review]
也要解下衝突；

Line app 那邊可以開兩個，一個 local 一個 production

File: pages/api/line/events.ts (hunk 2 above)
Comment (moojing) [reply pages/api/line/events.ts]: done https://github.com/Yukaii/zuzugo/pull/89/commits/f5aab95b16ccc45549bd0e295a70913fe57804b3

File: lib/notification/lineBot.ts (hunk 1 above)
Comment (moojing) [reply lib/notification/lineBot.ts]: https://github.com/Yukaii/zuzugo/pull/89/commits/f5aab95b16ccc45549bd0e295a70913fe57804b3

File: lib/notification/lineBot.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Client, TextMessage } from '@line/bot-sdk';
+
+// set Channel Access Token and Channel Secret for Line Bot 
+const client = new Client({
+  channelAccessToken: process.env.LINE_ACCESS_TOKEN || '',
```
Comment (Yukaii) [inline lib/notification/lineBot.ts]: env 也進 config.ts 吧~~可以弄成 scoped 的：

```ts
line: {
  accessToken: process.env.xxxxxx
}
```

File: pages/api/line/events.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    const {
+      userId = '',
+      groupId = '',
+      roomId = '',
+      // @todo: save chatType to database.
+      // type: chatType,
+    } = firstEvent.source as User & Group & Room;
+
+    const chatId = userId || groupId || roomId;
+    await sendLineBotMessage(chatId)
```
Comment (Yukaii) [inline pages/api/line/events.ts]: 這邊要 try catch 個？

**Review by Yukaii — CHANGES_REQUESTED** [review]

File: lib/notification/lineBot.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// @todo: pass rent data from api.
+const message: TextMessage = {
+  type: 'text',
+  text: 'Zuzugo got a new house!',
+};
+
+// @todo: get groupId from database.
+export default async function sendLineBotMessage(groupId: string) {
+  await client.pushMessage(groupId, message)
+    .then(() => {
```
Comment (Yukaii) [inline lib/notification/lineBot.ts]: 噢發現你這邊就 catch 了，還是讓 error 浮到上層好了，這邊的 then 和 catch 就拿掉

**Review by Yukaii — CHANGES_REQUESTED** [review]

File: lib/notification/lineBot.ts (hunk 5 above)
Comment (moojing) [reply lib/notification/lineBot.ts]: 對椰

**Review by Yukaii — APPROVED** [review]
### **ドーンだYO!!**

img

**Pull request merged**